greenpithumb/pump_scheduler.py

```diff
@@ -0,0 +1,36 @@
+import clock
+import tzlocal
+
+
+class PumpScheduler(object):
+    """A wrapper for managing pump scheduling."""
+
+    def __init__(self, clock, sleep_windows_local):
+        """Creates new PumpScheduler wrapper.
+        
+        Args:
+            clock: A clock interface
+            sleep_windows_local: A tuple of tuples, each consisting of a sleep 
+                window.
+        """
+        self._clock = clock
+        self._sleep_windows_local = sleep_windows_local
+
+    def is_running_pump_allowed(self):
+        """Returns True if OK to run pump, otherwise False."""
+        time_utc = self._clock.now()
+        time_local = time_utc.astimezone(tzlocal.get_localzone())
+        hour = time_local.hour
+
+        for window in self._sleep_windows_local:
+            window[0] = sleep_hour
+            window[1] = wake_hour
```

Comment:
This looks backwards since we want to assign the other way around. There's a neat way of doing this easily in python like:

```
for sleep_hour, wake_hour in self._sleep_windows_local:
```